TCPclient.cpp

```diff
@@ -3,36 +3,55 @@
 
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <unistd.h> //included this for the close function
+#define _XOPEN_SOURCE_EXTENDED 1
 
-#include <netinet/in.h>  
+#include <netinet/in.h> //contains socket structs
 
-int main() {
+int main()
+{
 
- 	//creating client socket 
+	// creating client socket
 	int network_socket;
-	network_socket = socket(AF_INET, SOCK_STREAM, 0)
-
-	// specify an address for the socket 
-	struct sockaddr_in server_address; 
-	server_address.sin_family = AF_INET;    //what type of address?
-	server_address.sin_port = htons(9002); 
+	network_socket = socket(AF_INET, SOCK_STREAM, 0);
+
+	// specify an address for the socket to connect to
+	struct sockaddr_in server_address;
+	// need to define a few fields within the sockaddr_in struct
+	// what is the family
+	server_address.sin_family = AF_INET;
+	// specify port
+	// htons = host to network (making sure data format is correct)
+	// puts integer port into correct byte order
+	// pass the actual port, htons takes care of conversion
+	server_address.sin_port = htons(9002);
```

Comment:
Why port 9002.  Make 9002 a varible,  maybe a const static?

Reply:
use google flags,  do error checking on the input before you assign it to the struct